### ISSUE TITLE
Native support to generate/parse SDP records.

### ISSUE THREAD

**Issue by jrosssavant** [EricssonResearch/openwebrtc#147]
The JSEP draft spec establishes SDP as the session description format.  As far as I can tell, the existing OpenWebRTC SDP record generation & parsing is done in Javascript and is not readily available to native clients.  I suggest utilities be added to generate an SDP record from a local session, and parse an sdp record into a remote session.

Thanks,
Joe

**Comment by ikonst** [comment]
Are we sure we want to perform string processing in native code? I thought it rather elegant that it was delegated to JS.

**Comment by jrosssavant** [comment]
Well, the native client needs to be able to talk to a web client on the other end.  They need to exchange the session description in a consistent format.  So the native code needs to be able to generate/parse the SDP record.  I think it is fine if that is done via a native-invoked eval of the JS code, but it would be great if it were simple to call from the native code.  Maybe this is easy to do but if so, it's not obvious to me.

**Comment by stefanalund** [comment]
Similar discussion here https://github.com/EricssonResearch/openwebrtc/pull/129

**Comment by jrosssavant** [comment]
There is a similar large volume of Java code included in the Android NativeCall app to assemble the JSON:

https://github.com/Rugvip/openwebrtc-examples/commit/dce2e8e38aff53a856ee3886dc1625840a00d444

**Comment by sdroege** [comment]
Note that GStreamer has a SDP parsing/generation library already, libgstsdp from gst-plugins-base.

**Comment by jrosssavant** [comment]
Yes, I spotted that earlier and have added it to my list of things to check out.  Thanks!

**Comment by superdump** [comment]
@stefanalund has now got some code working in iOS that parses the SDP using sdp.js and a JSContext and then gives an NSDictionary back. I expect similar things can be done on other platforms, the main issue is what to do in native C/GObject code.

**Comment by sdroege** [comment]
What's missing in libgstsdp that would be required or useful here?

**Comment by superdump** [comment]
We would have to check. It's a possibility...

**Comment by superdump** [comment]
I think one suggestion for this would be to use the GStreamer SDP parsing code as part of a set of objects in OpenWebRTC that handle creation of those objects that are used by sessions. So for example something that takes an SDP and spits out the payloads and other bits and pieces needed to configure sessions. And then similarly something that can be configured and spit out an SDP.

**Comment by suganthikarthick** [comment]
We have written our own implementation for sdp pasring to json and vice versa.
Now, we need a way to generate SDP offer to be done in C, using openwebrtc.

Thanks.

**Comment by superdump** [comment]
You should be able to see in the JavaScript bridge code (https://github.com/EricssonResearch/openwebrtc/blob/master/bridge/client/webrtc.js#L449) how this is done. You basically need information about sessions, payloads and ICE candidates that you know that you support.

**Comment by suganthikarthick** [comment]
Ok. We will have a look into that. Thanks.

**Comment by suganthikarthick** [comment]
Hi,

I gone through the ios sdk for openWebRTCNativeHandler, where an example is provided for creating the offer. Thank you.
